opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -1428,29 +1430,63 @@ namespace Opm {
             }
 
             const size_t max_num_cells_faillog = 20;
-            if (failed_cells_pb.size() > 0) {
+
+            int pb_size = failed_cells_pb.size(), pd_size = failed_cells_pd.size();
+            std::vector<int> displ_pb, displ_pd, recv_len_pb, recv_len_pd;
+            const auto& comm = grid_.comm();
+
+            if ( comm.rank() == 0 )
+            {
+                displ_pb.resize(comm.size()+1, 0);
+                displ_pd.resize(comm.size()+1, 0);
+                recv_len_pb.resize(comm.size());
+                recv_len_pd.resize(comm.size());
+            }
+
+            comm.gather(&pb_size, recv_len_pb.data(), 1, 0);
```

Comment:
The process of gathering a variable number of items from each rank is done in a quite low-level way here. Perhaps it would be good to have higher level class for this in the future? I think it looks correct, but a separate class would be possible to unit-test etc.